### PR TITLE
Use addr_of! macro instead of &

### PR DESCRIPTION
**Pull request by OuyangHang33**
Fix: #709 

General Clean up.

### CODE REVIEW

**Comment by OuyangHang33** [comment]
@vijaydhanraj Could you help to review this PR? Thanks!

**Review by jyao1 — APPROVED** [review]

File: tdx-tdcall/src/tdx.rs
Comment (vijaydhanraj) [inline tdx-tdcall/src/tdx.rs:1]: Thanks Hank! Can addr_of! be used here as well? Otherwise LGTM.
`let buffer: u64 = &digest.data as *const u8 as u64;`

Comment (OuyangHang33) [reply tdx-tdcall/src/tdx.rs:1]: Sure, it also need to change. I will update it

**Review by vijaydhanraj — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**